### PR TITLE
Dask bag migration

### PR DESCRIPTION
**Pull request by cowlicks**
Brings in stuff from `dask.bag.into` and its tests. I still need to add `costs` to the conversions here.

### CODE REVIEW

File: odo/backends/tests/test_dask.py [hunk 1]
```diff
@@ -3,14 +3,17 @@
 import pytest
 pytest.importorskip('dask')
```
Comment (mrocklin) [inline odo/backends/tests/test_dask.py]: Should also skip on `dask.bag`.  It has some dependencies that aren't explicitly stated in dask proper.

**Comment by mrocklin** [comment]
This seems fine to me.

File: odo/backends/tests/test_dask.py [hunk 2]
```diff
@@ -2,15 +2,19 @@
 
 import pytest
 pytest.importorskip('dask')
+pytest.importorskip('dask.bag')
```
Comment (cowlicks) [inline odo/backends/tests/test_dask.py]: @mrocklin like this? I'm not sure what the build env would look like to cause that.

Comment (mrocklin) [inline odo/backends/tests/test_dask.py]: Yes, like that.

At the moment `dask` does not list all of the dependencies of its subprojects.  If one `pip` or `conda` installs dask you don't necessarily get all of `dill`, `numpy`, `bcolz` required for each of `dask.bag`, `dask.array`, `dask.frame` respectively.  Importing dask doesn't necessarily import all of the subprojects though.  Users will get the `ImportError` only if they try to import a subproject for which their environment does not contain the necessary dependencies.

**Comment by cpcloud** [comment]
@mrocklin @cowlicks is this ready for merging?

**Comment by mrocklin** [comment]
+1

**Comment by cowlicks** [comment]
Thanks!

**Pull request merged**